geomstats/geometry/connection.py

```diff
@@ -458,7 +458,90 @@ def curvature(self, tangent_vec_a, tangent_vec_b, tangent_vec_c, base_point):
         curvature : array-like, shape=[..., {dim, [n, m]}]
             Tangent vector at `base_point`.
         """
-        raise NotImplementedError("The curvature is not implemented.")
+        riemann = self.riemann_tensor(base_point)
+        curvature = gs.einsum(
+            "...ijkl, ...i, ...j, ...k -> ...l",
+            riemann,
+            tangent_vec_a,
+            tangent_vec_b,
+            tangent_vec_c,
+        )
+        return curvature
+
+    def riemann_tensor(self, base_point):
+        r"""Compute Riemannian tensor at base_point.
+
+        Parameters
+        ----------
+        base_point :  array-like, shape=[..., {dim, [n, m]}]
+            Point on the group. Optional, default is the identity.
+
+        Returns
+        -------
+        riemann_curvature : array-like, shape=[..., {dim, [n, m]}, {dim, [n, m]},
+                                                    {dim, [n, m]}, {dim, [n, m]}]
+            R_ijk^l = riemann_tensor[i,j,k,l]
+            Riemannian tensor curvature.
+        """
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
```

Comment:
Does this work when `base_point` is a matrix?

Can we somehow avoid its use? Because we create a new array every time we do it. `christoffels` should work with a base point or multiple base points, no?

Reply:
Yes I was using it to know the number of points in base_point (to be able to permute the axes using gs.transpose), but if gs.move_axis (which I did not know existed) works I will get rid of these lines.

Reply:
But if `base_point` is a matrix (representing only a point in the manifold), you will get wrong information, no?

Your branch is too far behind, that's why you cannot find `moveaxis`. Please merge master after pulling upstream.

Reply:
It works apparently because there is a squeeze in `christoffels`. So for `base_point` of shape (1,dim), `christoffels` will return shape (dim, dim, dim) and `jacobian_christoffels` (dim, dim, dim, dim) because it is defined with a squeeze. So these lines are really just to get the number of points in `base_point`. I hope this is what you are talking about. This is a minimum reproducible example showing it works for several types of `base_point` (and namely a matrix representing only a point in the manifold).
```
import geomstats.backend as gs
from geomstats.information_geometry.dirichlet import DirichletDistributions

space = DirichletDistributions(3)

point_1 = space.random_point()
point_2 = gs.array([[1.0, 1.0, 1.0]])
point_3 = gs.array([[1.0, 2.0, 2.0], [1.0, 2.0, 1.0]])

print(space.metric.riemann_tensor(point_1))
print(space.metric.riemann_tensor(point_2))
print(space.metric.riemann_tensor(point_3))
```

Ok I will merge thanks!
Do you know if the methods I added in `data_generation` should rather be in `connection_data` and `riemannian_metric_data`? That way I won't have to rewrite the data generation for every manifold?

Reply:
So turns out I am keeping `base_point = gs.to_ndarray(base_point, to_ndim=2)` in my last commit also because I want to compute the jacobian christoffels for each of the points in `base_point`, which is not possible otherwise (or so far I haven't found anything).